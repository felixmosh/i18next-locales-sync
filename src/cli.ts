#!/usr/bin/env node

import fs from 'fs-extra';
import * as path from 'path';
import yargs from 'yargs';
import { LIB_PREFIX } from './constats';
import chalk from 'chalk';
import { syncLocales } from './index';

const options = yargs.usage('i18next-locales-sync -p en -s de ja he -l ./path/to/locales').option({
  primaryLanguage: {
    alias: 'p',
    type: 'string',
    description: 'The primary (source) language',
    default: 'en',
  },
  secondaryLanguages: {
    alias: 's',
    description: 'A list of all other supported languages',
    type: 'array',
    default: [],
  },
  localesFolder: {
    alias: 'l',
    description: 'The locals folder path (can be relative)',
    type: 'string',
    normalize: true,
  },
  outputFolder: {
    alias: 'o',
    description: 'The output folder',
    defaultDescription: '`localesFolder`',
    type: 'string',
    normalize: true,
  },
  config: {
    alias: 'c',
    description: 'A path to the config file',
    type: 'string',
    normalize: true,
  },
  useEmptyString: {
    alias: 'e',
    description: 'Use empty string as a value for new keys',
    type: 'boolean',
    normalize: true,
    default: false,
    defaultDescription: '`false`',
  },
}).argv;

if (options.config) {
  options.config = path.resolve(options.config);
  if (!fs.existsSync(options.config)) {
    throw new Error(chalk.red`${LIB_PREFIX} Config file '${options.config}' doesn't exist`);
  }

  const usersOptions = require(options.config);

  Object.assign(options, usersOptions);
}

if (!options.localesFolder) {
  throw new Error(chalk.red`${LIB_PREFIX} 'localesFolder' is mandatory option`);
} else {
  options.localesFolder = path.resolve(options.localesFolder);

  if (!fs.existsSync(options.localesFolder)) {
    throw new Error(
      chalk.red`${LIB_PREFIX} Locales folder '${options.localesFolder}' doesn't exist`
    );
  }
}

if (options.outputFolder) {
  options.outputFolder = path.resolve(options.outputFolder);
}

syncLocales({
  primaryLanguage: options.primaryLanguage,
  secondaryLanguages: options.secondaryLanguages,
  localesFolder: options.localesFolder,
  outputFolder: options.outputFolder,
  overridePluralRules: options.overridePluralRules as any,
  useEmptyString: options.useEmptyString,
});
