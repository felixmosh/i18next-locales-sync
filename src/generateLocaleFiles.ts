import crypto from 'crypto';
import fs from 'fs-extra';
import { fdir } from 'fdir';
import picomatch from 'picomatch';
import path from 'path';
import { LocaleFile, LocalesFiles } from '../types/types';
import { LIB_PREFIX } from './constats';

interface Options {
  primaryLanguage: string;
  otherLanguages: string[];
  localesFolder: string;
  fileExtension: string;
}

function extractLanguagesFromPath(filepath: string, allLanguages: string[]) {
  const pathParts = filepath.split(/[\\/]/g);
  return allLanguages.find((language) => pathParts.some((part) => part === language));
}

function extractNamespaceFromPath(filepath: string, language: string) {
  const pathParts = filepath.split(/[\\/]/g);

  if (pathParts.length < 2) {
    // handle empty namespace
    pathParts.push('');
  }

  const namespaceParts = pathParts.filter((part) => part !== language);

  return namespaceParts.join('/');
}

function addMissingLanguages(localeFiles: LocalesFiles, otherLanguages: string[]) {
  otherLanguages.filter((lang) => !localeFiles[lang]).forEach((lang) => (localeFiles[lang] = {}));

  return localeFiles;
}

function addMissingNamespaces(
  localeFiles: LocalesFiles,
  primaryLanguage: string,
  otherLanguages: string[]
) {
  const primaryFiles = localeFiles[primaryLanguage];

  Object.keys(primaryFiles).forEach((namespace) => {
    otherLanguages.forEach((otherLanguage) => {
      const filePath = primaryFiles[namespace].filePath
        .split(/[\\/]/g)
        .map((part) => {
          if (part.startsWith(primaryLanguage)) {
            return part.replace(primaryLanguage, otherLanguage);
          }
          return part;
        })
        .join(path.sep);

      localeFiles[otherLanguage][namespace] = localeFiles[otherLanguage][namespace] || {
        filePath,
        hash: '',
        data: {},
      };
    });
  });
}

export function populateFromDisk(filePath: string): LocaleFile {
  let data = {};
  let hash = '';

  if (fs.existsSync(filePath)) {
    data = fs.readJSONSync(filePath, { encoding: 'utf8' });

    hash = crypto
      .createHash('md5')
      .update(JSON.stringify(data))
      .digest('hex');
  }

  return { filePath, data, hash };
}

export function generateLocaleFiles({
  localesFolder,
  fileExtension,
  primaryLanguage,
  otherLanguages,
}: Options): LocalesFiles {
  const matcher = picomatch(`**/*${fileExtension}`);
  const paths = new fdir()
    .withFullPaths()
    .filter((path) => matcher(path))
    .crawl(localesFolder)
    .sync();
  const allLanguages = [primaryLanguage].concat(otherLanguages);

  const localeFiles = paths.reduce((structure, filePath) => {
    const remainingPath = path.relative(localesFolder, filePath);
    const filename = path.basename(remainingPath, fileExtension);
    const filepathWithoutExtension = path.join(path.dirname(remainingPath), filename);
    const language = extractLanguagesFromPath(filepathWithoutExtension, allLanguages);

    if (!language) {
      return structure;
    }

    const namespace = extractNamespaceFromPath(filepathWithoutExtension, language);

    structure[language] = structure[language] || {};
    structure[language][namespace] = populateFromDisk(filePath);

    return structure;
  }, {} as LocalesFiles);

  if (!localeFiles[primaryLanguage]) {
    throw new Error(
      `${LIB_PREFIX} There are no files for your primary language (${primaryLanguage})`
    );
  }

  addMissingLanguages(localeFiles, otherLanguages);
  addMissingNamespaces(localeFiles, primaryLanguage, otherLanguages);

  return localeFiles;
}
