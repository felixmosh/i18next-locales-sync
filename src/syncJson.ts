import { JSONObject, JSONValue, LocaleObject } from '../types/types';
import { MAX_DEPTH } from './constats';
import { PluralResolver } from './i18next/PluralResolver';
import { hasSomePluralSuffix } from './utils/hasSomePluralSuffix';
import { isObject } from './utils/isObject';
import { traverse } from './utils/traverse';

function generatePluralForms(
  {
    sourceKey,
    sourceLng,
    targetLng,
    newTargetObject,
    targetObject,
    sourceObject,
  }: {
    sourceObject: JSONObject;
    targetObject: JSONObject;
    newTargetObject: JSONObject;
    sourceKey: string;
    sourceLng: string;
    targetLng: string;
  },
  pluralResolver: PluralResolver,
  useEmptyString = false
) {
  const singularSourceKey = pluralResolver.getSingularFormOfKey(sourceKey, sourceLng);

  const pluralForms = pluralResolver.getPluralFormsOfKey(singularSourceKey, targetLng);
  const fallbackValue = useEmptyString ? '' : sourceObject[sourceKey];

  pluralForms.forEach((key) => {
    newTargetObject[key] =
      (targetObject && targetObject[key] && !isObject(targetObject[key])
        ? targetObject[key]
        : useEmptyString
        ? ''
        : sourceObject[key]) || fallbackValue;
  });
}

function syncEntry({
  sourceLng,
  targetLng,
  pluralResolver,
  useEmptyString,
}: {
  sourceLng: string;
  targetLng: string;
  pluralResolver: PluralResolver;
  useEmptyString?: boolean;
}) {
  const sourceSuffixes = pluralResolver.getPluralFormsOfKey('', sourceLng).filter(Boolean);
  const targetSuffixes = pluralResolver.getPluralFormsOfKey('', targetLng).filter(Boolean);

  const isTargetRequiresPluralForm = pluralResolver.needsPlural(targetLng);

  return (
    sourceObject: JSONObject,
    targetObject: JSONObject,
    newTargetObject: JSONObject,
    sourceKey: string,
    sourceValue: JSONValue,
    targetValue: JSONValue
  ) => {
    if (Array.isArray(sourceValue)) {
      newTargetObject[sourceKey] = [];
    } else if (isObject(sourceValue)) {
      newTargetObject[sourceKey] = {};
    } else if (hasSomePluralSuffix(sourceKey, sourceSuffixes)) {
      isTargetRequiresPluralForm &&
        generatePluralForms(
          {
            sourceObject,
            targetObject,
            newTargetObject,
            sourceKey,
            sourceLng,
            targetLng,
          },
          pluralResolver,
          useEmptyString
        );
    } else {
      newTargetObject[sourceKey] =
        targetValue && !isObject(targetValue) ? targetValue : useEmptyString ? '' : sourceValue;

      // keeps existing plural forms
      targetSuffixes.forEach((suffix) => {
        const pluralTargetKey = `${sourceKey}${suffix}`;

        if (targetObject && targetObject[pluralTargetKey]) {
          newTargetObject[pluralTargetKey] = targetObject[pluralTargetKey];
        }
      });
    }
  };
}

interface SyncJsonOptions {
  source: LocaleObject;
  target: LocaleObject;
  pluralResolver: PluralResolver;
  depth?: number;
  useEmptyString?: boolean;
}

export function syncJson({
  source,
  target,
  pluralResolver,
  depth = MAX_DEPTH,
  useEmptyString = false,
}: SyncJsonOptions) {
  target.data = traverse(
    source.data,
    target.data,
    syncEntry({
      sourceLng: source.language,
      targetLng: target.language,
      pluralResolver,
      useEmptyString,
    }),
    depth
  );

  return target;
}
