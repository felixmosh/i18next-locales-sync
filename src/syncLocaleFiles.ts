import { LocalesFiles } from '../types/types';
import { PluralResolver } from './i18next/PluralResolver';
import { syncJson } from './syncJson';

interface Options {
  localeFiles: LocalesFiles;
  primaryLanguage: string;
  otherLanguages: string[];
  pluralResolver: PluralResolver;
  useEmptyString?: boolean;
}

export function syncLocaleFiles({
  localeFiles,
  primaryLanguage,
  otherLanguages,
  pluralResolver,
  useEmptyString,
}: Options): LocalesFiles {
  const primaryLocaleFiles = localeFiles[primaryLanguage];
  otherLanguages.forEach((currentLanguage) => {
    const currentNamespaces = localeFiles[currentLanguage];

    Object.keys(primaryLocaleFiles).forEach((primaryNamespace) => {
      const { data } = syncJson({
        source: { data: primaryLocaleFiles[primaryNamespace].data, language: primaryLanguage },
        target: { data: currentNamespaces[primaryNamespace].data, language: currentLanguage },
        pluralResolver,
        useEmptyString,
      });

      currentNamespaces[primaryNamespace].data = data;
    });
  });

  return localeFiles;
}
