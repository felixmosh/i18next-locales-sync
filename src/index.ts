import chalk from 'chalk';
import { LIB_PREFIX } from './constats';
import { generateLocaleFiles } from './generateLocaleFiles';
import { PluralResolver } from './i18next/PluralResolver';
import { syncLocaleFiles } from './syncLocaleFiles';
import { writeToDisk } from './utils/writeToDisk';

interface SyncLocalesOptions {
  primaryLanguage: string;
  secondaryLanguages: string[];
  localesFolder: string;
  outputFolder?: string;
  fileExtension?: string;
  overridePluralRules?: (pluralResolver: PluralResolver) => PluralResolver;
  useEmptyString?: boolean;
}

export function syncLocales({
  primaryLanguage,
  secondaryLanguages: otherLanguages,
  localesFolder,
  outputFolder = localesFolder,
  overridePluralRules,
  fileExtension = '.json',
  useEmptyString = false,
}: SyncLocalesOptions) {
  const pluralResolver = new PluralResolver();

  if (typeof overridePluralRules === 'function') {
    overridePluralRules(pluralResolver);
  }

  const localeFiles = generateLocaleFiles({
    primaryLanguage,
    otherLanguages,
    localesFolder,
    fileExtension,
  });

  syncLocaleFiles({
    localeFiles,
    primaryLanguage,
    otherLanguages,
    pluralResolver,
    useEmptyString,
  });

  writeToDisk({
    localeFiles,
    primaryLanguage,
    otherLanguages,
    outputFolder,
    localesFolder,
  });

  console.log(
    chalk.green`${chalk.bold.greenBright(LIB_PREFIX)} '${localesFolder}' were synced successfully.`
  );
}
