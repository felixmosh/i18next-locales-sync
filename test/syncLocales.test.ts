import { DirectoryJSON, vol } from 'memfs';
import { relative, resolve } from 'path';
import { syncLocales } from '../src';

function makePathNoneUnique(fileSystemList: DirectoryJSON) {
  return Object.keys(fileSystemList).reduce((result, filePath) => {
    const noneUniquePath = relative(resolve('.'), filePath).replace(/[\\]/g, '/');
    result[noneUniquePath] = fileSystemList[filePath];
    return result;
  }, {} as any);
}

describe('syncLocales - E2E', () => {
  it('should sync locale files without namespaces', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const outputFolder = resolve('./test/output/fixture1');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: resolve('./test/fixtures/fixture1'),
      outputFolder,
      fileExtension: '.json',
    });

    expect(makePathNoneUnique(vol.toJSON(outputFolder))).toMatchSnapshot();
  });

  it('should sync locale files with namespaces', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const outputFolder = resolve('./test/output/fixture2');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: resolve('./test/fixtures/fixture2'),
      outputFolder,
      fileExtension: '.json',
    });

    expect(makePathNoneUnique(vol.toJSON(outputFolder))).toMatchSnapshot();
  });

  it('should sync locale files with nested namespaces', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const outputFolder = resolve('./test/output/fixture3');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: resolve('./test/fixtures/fixture3'),
      outputFolder,
      fileExtension: '.json',
    });

    expect(makePathNoneUnique(vol.toJSON(outputFolder))).toMatchSnapshot();
  });

  it('should sync locales to the same locales folder', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const localesFolder = resolve('./test/fixtures/fixture3');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: localesFolder,
      outputFolder: localesFolder,
      fileExtension: '.json',
    });

    expect(makePathNoneUnique(vol.toJSON(localesFolder))).toMatchSnapshot();
  });

  it('should sync locale files with different folder structure', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const outputFolder = resolve('./test/output/fixture4');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: resolve('./test/fixtures/fixture4'),
      outputFolder,
      fileExtension: '.json',
    });

    expect(makePathNoneUnique(vol.toJSON(outputFolder))).toMatchSnapshot();
  });

  it('should sync locale files with custom spaces', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const outputFolder = resolve('./test/output/with-spaces');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: resolve('./test/fixtures/fixture5'),
      outputFolder,
      fileExtension: '.json',
      spaces: 4,
    });

    expect(makePathNoneUnique(vol.toJSON(outputFolder))).toMatchSnapshot();
  });

  it('should sync locale files with deprecated compatibilityJSON', () => {
    const primaryLanguage = 'en';
    const otherLanguages = ['ja', 'he', 'de'];
    const outputFolder = resolve('./test/output/fixture5');

    syncLocales({
      primaryLanguage,
      secondaryLanguages: otherLanguages,
      localesFolder: resolve('./test/fixtures/fixture5'),
      outputFolder,
      fileExtension: '.json',
      compatibilityJSON: 'v3',
    });

    expect(makePathNoneUnique(vol.toJSON(outputFolder))).toMatchSnapshot();
  });
});
