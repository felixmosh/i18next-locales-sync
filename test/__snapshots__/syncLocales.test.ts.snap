// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`syncLocales - E2E should sync locale files when using override plural rules 1`] = `
Object {
  "test/output/with-override-plural-rules/ar/common.json": "{
  \\"test\\": \\"bla-ar\\",
  \\"test_plural\\": \\"bla-ar-plural\\"
}",
  "test/output/with-override-plural-rules/he/common.json": "{
  \\"test\\": \\"bla-he\\",
  \\"test_plural\\": \\"bla-he-plural\\"
}",
}
`;

exports[`syncLocales - E2E should sync locale files with custom spaces 1`] = `
Object {
  "test/output/with-spaces/common/de.json": "{
    \\"test\\": \\"bla-en\\",
    \\"test_plural\\": \\"bla-plural-en\\"
}",
  "test/output/with-spaces/common/en.json": "{
    \\"test\\": \\"bla-en\\",
    \\"test_plural\\": \\"bla-plural-en\\"
}",
  "test/output/with-spaces/common/he.json": "{
    \\"test\\": \\"bla-he\\",
    \\"test_plural\\": \\"bla-plural-en\\"
}",
  "test/output/with-spaces/common/ja.json": "{
    \\"test\\": \\"bla-en\\",
    \\"test_plural\\": \\"bla-plural-en\\"
}",
  "test/output/with-spaces/front/de.json": "{
    \\"test\\": \\"bla-en\\"
}",
  "test/output/with-spaces/front/en.json": "{
    \\"test\\": \\"bla-en\\"
}",
  "test/output/with-spaces/front/he.json": "{
    \\"test\\": \\"bla\\"
}",
  "test/output/with-spaces/front/ja.json": "{
    \\"test\\": \\"bla-en\\"
}",
}
`;

exports[`syncLocales - E2E should sync locale files with deprecated compatibilityJSON 1`] = `
Object {
  "test/output/fixture5/common/de.json": "{
  \\"test\\": \\"bla-en\\",
  \\"test_plural\\": \\"bla-plural-en\\"
}",
  "test/output/fixture5/common/en.json": "{
  \\"test\\": \\"bla-en\\",
  \\"test_plural\\": \\"bla-plural-en\\"
}",
  "test/output/fixture5/common/he.json": "{
  \\"test\\": \\"bla-he\\",
  \\"test_1\\": \\"bla-1-he\\",
  \\"test_0\\": \\"bla-plural-en\\",
  \\"test_2\\": \\"bla-plural-en\\",
  \\"test_3\\": \\"bla-plural-en\\"
}",
  "test/output/fixture5/common/ja.json": "{
  \\"test\\": \\"bla-en\\"
}",
  "test/output/fixture5/front/de.json": "{
  \\"test\\": \\"bla-en\\"
}",
  "test/output/fixture5/front/en.json": "{
  \\"test\\": \\"bla-en\\"
}",
  "test/output/fixture5/front/he.json": "{
  \\"test\\": \\"bla\\",
  \\"test_0\\": \\"bla-0\\"
}",
  "test/output/fixture5/front/ja.json": "{
  \\"test\\": \\"bla-en\\"
}",
}
`;

exports[`syncLocales - E2E should sync locale files with different folder structure 1`] = `
Object {
  "test/output/fixture4/common/de.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture4/common/en.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture4/common/he.json": "{
  \\"test\\": \\"bla-he\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_two\\": \\"bla other en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture4/common/ja.json": "{
  \\"test\\": \\"bla en\\"
}",
  "test/output/fixture4/front/de.json": "{
  \\"test\\": \\"bla-en\\"
}",
  "test/output/fixture4/front/en.json": "{
  \\"test\\": \\"bla-en\\"
}",
  "test/output/fixture4/front/he.json": "{
  \\"test\\": \\"bla\\"
}",
  "test/output/fixture4/front/ja.json": "{
  \\"test\\": \\"bla-en\\"
}",
}
`;

exports[`syncLocales - E2E should sync locale files with namespaces 1`] = `
Object {
  "test/output/fixture2/de/common.json": "{
  \\"test\\": \\"bla-de\\",
  \\"test_one\\": \\"bla-de\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture2/de/front.json": "{
  \\"title\\": \\"front\\"
}",
  "test/output/fixture2/en/common.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture2/en/front.json": "{
  \\"title\\": \\"front\\"
}",
  "test/output/fixture2/he/common.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_two\\": \\"bla other en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture2/he/front.json": "{
  \\"title\\": \\"front\\"
}",
  "test/output/fixture2/ja/common.json": "{
  \\"test\\": \\"bla en\\"
}",
  "test/output/fixture2/ja/front.json": "{
  \\"title\\": \\"front\\"
}",
}
`;

exports[`syncLocales - E2E should sync locale files with nested namespaces 1`] = `
Object {
  "test/output/fixture3/de/common.json": "{
  \\"test\\": \\"bla-de\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture3/de/nested/a.json": "{
  \\"a\\": \\"bla-de\\"
}",
  "test/output/fixture3/de/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
  "test/output/fixture3/en/common.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture3/en/nested/a.json": "{
  \\"a\\": \\"bla-en\\"
}",
  "test/output/fixture3/en/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
  "test/output/fixture3/he/common.json": "{
  \\"test\\": \\"bla-he\\",
  \\"test_two\\": \\"bla-two-he\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture3/he/nested/a.json": "{
  \\"a\\": \\"bla-en\\"
}",
  "test/output/fixture3/he/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
  "test/output/fixture3/ja/common.json": "{
  \\"test\\": \\"bla en\\"
}",
  "test/output/fixture3/ja/nested/a.json": "{
  \\"a\\": \\"bla-en\\"
}",
  "test/output/fixture3/ja/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
}
`;

exports[`syncLocales - E2E should sync locale files without namespaces 1`] = `
Object {
  "test/output/fixture1/de.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture1/en.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture1/he.json": "{
  \\"test\\": \\"bla\\",
  \\"test_one\\": \\"bla-0\\",
  \\"test_two\\": \\"bla other en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/output/fixture1/ja.json": "{
  \\"test\\": \\"bla en\\"
}",
}
`;

exports[`syncLocales - E2E should sync locales to the same locales folder 1`] = `
Object {
  "test/fixtures/fixture3/de/common.json": "{
  \\"test\\": \\"bla-de\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/fixtures/fixture3/de/nested/a.json": "{
  \\"a\\": \\"bla-de\\"
}
",
  "test/fixtures/fixture3/de/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
  "test/fixtures/fixture3/en/common.json": "{
  \\"test\\": \\"bla en\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}
",
  "test/fixtures/fixture3/en/nested/a.json": "{
  \\"a\\": \\"bla-en\\"
}
",
  "test/fixtures/fixture3/en/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}
",
  "test/fixtures/fixture3/he/common.json": "{
  \\"test\\": \\"bla-he\\",
  \\"test_two\\": \\"bla-two-he\\",
  \\"test_one\\": \\"bla one en\\",
  \\"test_other\\": \\"bla other en\\"
}",
  "test/fixtures/fixture3/he/nested/a.json": "{
  \\"a\\": \\"bla-en\\"
}",
  "test/fixtures/fixture3/he/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
  "test/fixtures/fixture3/ja/common.json": "{
  \\"test\\": \\"bla en\\"
}",
  "test/fixtures/fixture3/ja/nested/a.json": "{
  \\"a\\": \\"bla-en\\"
}",
  "test/fixtures/fixture3/ja/nested/b.json": "{
  \\"b\\": \\"bla-en\\"
}",
}
`;
